apps/client/src/pages/community/community-edit/community-edit.tsx

```diff
@@ -68,29 +59,81 @@ const CommunityEdit = () => {
     },
   });
 
-  const handlePutFeed = () => {
-    if (isDisabled || !category) {
-      return;
+  const { mutate: postImageUploadMutate } = useMutation({
+    ...MUTATION_QUERY_OPTIONS.POST_IMAGE(),
+  });
+
+  if (!feedDetailData) {
+    throw new Error(
+      '글 수정 페이지에서 원본 피드 데이터를 불러오지 못했습니다.',
+    );
+  }
+
+  const [title, setTitle] = useState(feedDetailData?.title || '');
+  const [content, setContent] = useState(feedDetailData?.content || '');
+  const [category, setCategory] = useState(
+    feedDetailData?.category?.category || '',
+  );
+  const [newImages, setNewImages] = useState<
+    { file: File; previewUrl: string }[]
+  >([]);
+  const [deletedImageIds, setDeletedImageIds] = useState<number[]>([]);
+  const [updatedImages, setUpdatedImages] = useState<
+    { id?: number; imageUrl: string; sequence: number }[]
+  >(
+    () =>
+      feedDetailData.imageUrl?.filter(isValidImage).map((img, index) => ({
+        id: img.imageId,
+        imageUrl: img.imageUrl,
+        sequence: index + 1,
+      })) ?? [],
+  );
+  const { isErrorState } = useLimitedInput(LIMIT_SHORT_TEXT, title.length);
+
+  const handleUploadNewImages = async () => {
+    if (newImages.length === 0) {
+      return [];
     }
-    //@TODO 타입 에러로 임시 작성해둠. 추후 구현 시 수정 필요
+
+    const data = await new Promise<{ presignedUrls: string[] }>(
+      (resolve, reject) =>
+        postImageUploadMutate(
+          newImages.map((item) => item.file.type),
+          { onSuccess: resolve, onError: reject },
+        ),
+    );
```

Comment:
`mutateAsync`를 사용하면 새로운 Promise 객체를 생성할 필요 없이
```suggestion
    const data = await postImageUploadMutateAsync(
          newImages.map((item) => item.file.type),
        );
```
이렇게 변경할 수 있을것 같아요

Reply:
오 이 부분은 참고할만한 것 같아서 메모해두고, 추후에 알아볼게요